### ISSUE TITLE
[Feature] Reduce overfitting when training

### ISSUE THREAD

**Issue by adamltyson** [brainglobe/cellfinder#354]
Users are generating small amounts of data, and the network seems to be over fitting. They're having to manually get existing data to constrain their models. This could be improved by:
* Providing old training data via GIN
* Allowing upper layers to be [frozen](https://keras.io/getting-started/faq/#how-can-i-freeze-keras-layers)

Interested to know what aspects of training have helped @FedeClaudi @larsrollik

**Comment by matham** [comment]
To prevent overfitting with small data, in addition to freezing initial layers, there probably should probably also be some better learning rate schedulers. Such as [LearningRateScheduler](https://keras.io/api/callbacks/learning_rate_scheduler/) or even [ReduceLROnPlateau](https://keras.io/api/callbacks/reduce_lr_on_plateau/).

Slightly related, there probably should also be some input normalization step? Because different datasets have different ranges and I'm not sure transfer learning/fine tuning works well when train and tune datasets have wildly different mean/std. The way to go probably is to calculate the mean/std of the cubes when they are generated - i.e. add them to yaml, so you don't need to read over the dataset at the start every time you re-train.
